### PR TITLE
YCDOCIO-1597: Terraform configuration for Data Proc tutorial

### PR DESCRIPTION
**Pull request by maxdunaevsky-documentatio**
@redalka-cloud , привет! Прошу провести ревью.

[YCDOCIO-1597](https://st.yandex-team.ru/YCDOCIO-1597)

### CODE REVIEW

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 1]
```diff
@@ -0,0 +1,288 @@
+# Infrastructure for Yandex Cloud Managed Service for ClickHouse cluster, Data Proc cluster and virtual machine.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:1]: Infrastructure for Yandex Cloud Managed Service for the ClickHouse cluster, Data Proc cluster, and virtual machine.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+#
+# RU: https://cloud.yandex.ru/docs/data-proc/tutorials/exchange-data-with-mch
+#
+# Set the configuration of the Managed Service for ClickHouse cluster, Data Proc cluster and Virtual Machine.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:5]: Set the configuration of the Managed Service for ClickHouse cluster, Data Proc cluster, and Virtual Machine.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+
+# Specify the pre-installation parameters:
+locals {
+  # Base settings
+  folder_id  = ""              # Your folder ID.
+  network_id = ""              # Network ID for the Managed Service for ClickHouse cluster, the Data Proc cluster and VM.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:11]: Network ID for the Managed Service for ClickHouse cluster, the Data Proc cluster, and VM.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+  subnet_id  = ""              # Subnet ID (enable NAT for this subnet).
+  zone_id    = "ru-central1-a" # Availability zone for resources.
+
+  # Managed Service for ClickHouse cluster.
+  ch_user_name     = "user1" # Set user name for the ClickHouse cluster.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:16]: Set a username for the ClickHouse cluster.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+  ch_user_password = ""      # Set user password for the ClickHouse cluster.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:17]: Set a user password for the ClickHouse cluster.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+
+  # Data Proc cluster.
+  dp_ssh_key  = "" # Set absolute path to SSH public key for the Data Proc cluster.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:20]: Set an absolute path to the SSH public key for the Data Proc cluster.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+  dp_account  = "" # Name of the Data Proc cluster service account.
+  bucket_name = "" # Name for the Object Storage bucket. Should be unique in Cloud.
+
+  # VM
+  vm_username = ""                     # Set username for VM.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:25]: Set a username for VM.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+# Assign role `dataproc.agent` to the Data Proc cluster service account.
+resource "yandex_resourcemanager_folder_iam_binding" "dataproc-agent" {
+  folder_id = local.folder_id
+  role      = "dataproc.agent"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+# Assign role `dataprov.provisioner` to the Data Proc cluster service account.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:106]: Assign the `dataprov.provisioner` role  to the Data Proc cluster service account.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+resource "yandex_resourcemanager_folder_iam_binding" "dataproc-provisioner" {
+  folder_id = local.folder_id
+  role      = "dataproc.provisioner"
+  members = [
+    "serviceAccount:${yandex_iam_service_account.dataproc.id}",
+  ]
+}
+
+# Assign role `monitoring.viewer` to the Data Proc cluster service account.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:115]: Assign the `monitoring.viewer` role to the Data Proc cluster service account.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+    v4_cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+resource "yandex_iam_service_account" "dataproc" {
+  description = "Service account to manage the Data Proc cluster"
+  name        = local.dp_account
+}
+
+# Assign role `dataproc.agent` to the Data Proc cluster service account.
```
Comment (redalka-cloud) [inline tutorials/terraform/data-proc-data-exchange-with-mch.tf:97]: Assign the `dataproc.agent` role to the Data Proc cluster service account.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 1 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:1]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 3 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:11]: Оксфордская запятая? Исправил.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 4 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:16]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 5 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:17]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 6 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:20]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 7 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:25]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 8 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:106]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 9 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:115]: Исправлено.

File: tutorials/terraform/data-proc-data-exchange-with-mch.tf (hunk 10 above)
Comment (maxdunaevsky-documentatio) [reply tutorials/terraform/data-proc-data-exchange-with-mch.tf:97]: Исправлено.

**Pull request merged**